lib/galaxy/authnz/pkce_utils.py

```diff
@@ -0,0 +1,102 @@
+""" Sourced from https://github.com/RomeoDespres/pkce/blob/master/pkce/__init__.py
```

Comment:
Looks like this is published on pypi; we could add pkce it as a conditional dependency instead of including it directly. Might be worth swapping it over to get any updates automatically?  Or are there changes to this file from upstream?

Reply:
Yeah, I didn't make any real changes from the source, so that seems good.

Reply:
@gecage952 Want me to do it, or do you want to take a stab at it?  The rest of this looks great to me, thank you so much for the work here!

Reply:
It would probably be easier for you to do it, since you guys are more familiar with that process. Thanks.